[13.0][MIG] survey

Yasaie opened the pull request:
## Migration of `survey`

> Depends on `gamification`

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Yasaie committed:
[MIG] survey

gdgellatly commented:
Receive the following error migrating a database with only mis-builder as third party module.
```
psycopg2.IntegrityError: duplicate key value violates unique constraint "ir_model_fields_name_unique"
DETAIL:  Key (model, name)=(survey.question, survey_id) already exists.
```

gdgellatly commented on addons/survey/migrations/13.0.3.0/pre-migration.py:
```diff
@@ -0,0 +1,95 @@
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE survey_question sq
+        SET title = sp.title AND survey_id = sp.survey_id AND is_page = TRUE
```
incorrect syntax use `,` not `AND` to avoid

```
psycopg2.ProgrammingError: argument of AND must be type boolean, not type character varying
LINE 3:         SET title = sp.title AND survey_id = sp.survey_id AN...
```

gdgellatly commented on addons/survey/migrations/13.0.3.0/pre-migration.py:
```diff
@@ -0,0 +1,95 @@
+def update_survey_question_page_id(cr):
+    replaces = [
+        ('survey_user_input', 'last_displayed_page_id'),
+        ('survey_user_input_line', 'page_id'),
```
I think this field no longer exists, so causes error on update. Not sure, database I have has no records for this table.

Yasaie replied:
You were right. it's a related field now, thanks

MiquelRForgeFlow committed:
[IMP] survey migration

gdgellatly commented:
@MiquelRForgeFlow It might pay to check. Just testing a migration and the field access_token is being set on surveys as it is a new field, however all surveys are receiving an identical access token, which is then causing constraints not to be applied and singleton errors on accessing via respondent link.

pedrobaeza committed:
[FIX] survey: Generate unique access_token

Populate a unique token for each existing survey, as default method populates the same
for all.

Co-authored by: Graeme Gellatly <gdgellatly@gmail.com>
